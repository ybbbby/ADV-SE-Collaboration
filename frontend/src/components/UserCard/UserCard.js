import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Typography, Card, CardContent, Avatar } from '@material-ui/core'
import { withStyles } from '@material-ui/styles'
import { deepOrange } from '@material-ui/core/colors'

const styles = (theme) => ({
  root: {
    textAlign: 'center',
    borderRadius: 0,
    boxShadow:
      '0px 0px 0px -1px rgba(0,0,0,0.2), -2px -1px 1px -1px rgba(0,0,0,0.14), 0px 1px 4px 1px rgba(0,0,0,0.12);',
  },
  avatar: {
    margin: '10px auto',
    width: theme.spacing(9),
    height: theme.spacing(9),
  },
  username: {
    fontSize: 18,
  },
  email: {
    fontSize: 14,
  },
})

class UserCard extends Component {
  constructor(props) {
    super(props)

    this.state = {
      fullname: '',
      email: '',
      avatar: '',
    }
  }

  componentWillMount() {
    fetch('/userinfo', {
      method: 'GET',
    })
      .then((response) => response.json())
      .then((data) => {
        this.setState({
          fullname: data['name'],
          email: data['email'],
          avatar: data['picture'],
        })
      })
      .catch((error) => {
        console.error(error)
      })
  }

  render() {
    const { classes } = this.props
    return (
      <Card className={classes.root}>
        <CardContent>
          <Avatar className={classes.avatar} src={this.state.avatar} />
          <Typography className={classes.username}>
            {this.state.fullname}
          </Typography>
          <Typography className={classes.email} color="textSecondary">
            {this.state.email}
          </Typography>
        </CardContent>
      </Card>
    )
  }
}

UserCard.propTypes = {
  classes: PropTypes.object.isRequired,
}

export default withStyles(styles)(UserCard)
