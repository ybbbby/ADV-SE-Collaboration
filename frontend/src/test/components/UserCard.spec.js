/* eslint-env jest */
import { shallow } from 'enzyme'
import { HashRouter as Router } from 'react-router-dom'
import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import UserCard from '../../components/UserCard/UserCard'

const userDataLoggedIn = {
  name: 'test',
  email: 'test@columbia.edu',
  picture: 'testurl',
}

const userDataNotLoggedIn = {
  name: '',
  email: '',
  picture: '',
}

describe('UserCard', () => {
  test('renders user info correctly when logged in', () => {
    render(
      <Router>
        <UserCard data={userDataLoggedIn} isLogin={true} />
      </Router>
    )
    expect(screen.queryByText('test@columbia.edu')).toBeDefined()
    expect(screen.queryByText('test')).toBeDefined()
    expect(screen.queryByText('Create event')).toBeDefined()
  })

  test('renders login button when not logged in', () => {
    render(
      <Router>
        <UserCard data={userDataNotLoggedIn} isLogin={false} />
      </Router>
    )
    expect(screen.queryByText('Login')).toBeDefined()
  })

  test('renders login modal when clicks the login button', () => {
    render(
      <Router>
        <UserCard data={userDataNotLoggedIn} isLogin={false} />
      </Router>
    )
    userEvent.click(screen.getByRole('button'))
    expect(screen.findAllByRole('presentation')).toBeDefined()
  })

  test('close login modal when clicks the close button', async () => {
    render(
      <Router>
        <UserCard data={userDataNotLoggedIn} isLogin={false} />
      </Router>
    )
    userEvent.click(screen.getByRole('button'))
    userEvent.click(screen.getByRole('button', { name: 'close' }))
    // todo: no idea how to simulate
  })

  it('snapshot when logged in', () => {
    const wrapper = shallow(<UserCard data={userDataLoggedIn} isLogin={true} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('snapshot when not logged in', () => {
    const wrapper = shallow(
      <UserCard data={userDataNotLoggedIn} isLogin={false} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
