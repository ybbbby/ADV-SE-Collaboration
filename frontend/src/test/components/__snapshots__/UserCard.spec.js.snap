// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard snapshot when logged in 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Avatar))
    className="makeStyles-avatar-33"
    src="testurl"
  />
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-username-34"
  >
    test
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-email-35"
    color="textSecondary"
  >
    test@columbia.edu
  </WithStyles(ForwardRef(Typography))>
  <Link
    className="makeStyles-link-36"
    to="/newevent"
  >
    <WithStyles(ForwardRef(Button))
      className="makeStyles-createButton-37"
      color="primary"
    >
      Create event
    </WithStyles(ForwardRef(Button))>
  </Link>
</Fragment>
`;

exports[`UserCard snapshot when not logged in 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Avatar))
    className="makeStyles-avatar-33"
  />
  <WithStyles(ForwardRef(Button))
    color="inherit"
    onClick={[Function]}
  >
    Login
  </WithStyles(ForwardRef(Button))>
  <LoginModal
    handleClose={[Function]}
    open={false}
  />
</Fragment>
`;
